js/minesweeper.js

```diff
@@ -0,0 +1,210 @@
+    
+    for (var k = -1; k <= 1; k++) {
+      if (!field[cell.address.y + k]) {
+
+        continue;
+      }
+      for (var l = -1; l <= 1; l++) {
+        if (!field[cell.address.y + k][cell.address.x + l]) {
+
+          continue;
+        }
+        var currentCell = field[cell.address.y + k][cell.address.x + l];
+          if (currentCell.checked || currentCell.value === "mine") {
+
+            continue;
+        } else if (currentCell.value > 0 && currentCell.signed == false) {
+          currentCell.checked = true;
+          currentCell.opened = true;
+          currentCell.element.innerText = currentCell.value;
+          currentCell.element.classList.add("opened");
+        } else {
+          emptyCellsOpening(currentCell, field);
+        }
+      }
+    }
+  }
+}
+
+function cellClick (cell, field, closedCells) {
+	return function () {
+		if (cell.signed === true) {
+
+			return;
+		} else {
+			if (cell.value === "mine" && cell.opened === false) {
+				gameOver(field);
+				cell.element.classList.add("active-mine");
+			} else if (cell.value > 0 && cell.signed == false) {
+				cell.element.innerText = cell.value;
+				cell.element.classList.add("opened");
+				cell.checked = true;
+				cell.opened = true;
+				closedCells--;
+				isWon(field);
+			} else if (cell.signed == false) {
+				isWon(field);
+				emptyCellsOpening(cell, field);
+			}
+		}
+	}
+}
+
+function newGame () {
+	startGameDiv.classList.remove("hidden");
+	while(fieldElement.childNodes[0]) {
+		fieldElement.removeChild(fieldElement.childNodes[0]);
+	}
+	messageElement.innerText = "";
+	tryAgainButton.classList.add("hidden");
+}
+
+function endOfGame (field, result) {
+	messageElement.classList.add(result.messageClass);
+	messageElement.classList.remove("hidden");
+	messageElement.innerText = result.message;
+	tryAgainButton.classList.remove("hidden");
+	tryAgainButton.onclick = newGame;
+	countOfMines.classList.add("hidden");
+	for (var i = 0; i < field.length; i++) {
+		field[i].map(function (obj) {
+			if (obj.element && obj.value === "mine") {
+				obj.element.classList.add(result.mineCellClass);
+			} else if (obj.element && obj.value > 0) {
+				obj.element.innerText = obj.value;
+				obj.element.classList.add("opened");
+			} else if (obj.element) {
+				obj.element.classList.add("opened");
+			}
+			obj.opened = true;
+		});
+	}
+}
+
+function gameOver (field) {
+	var result = {
+		message: "game over!",
+		messageClass: "game-over",
+		mineCellClass: "mine"
+	}
+	endOfGame(field, result);
+}
+
+function isWon (field) {
+	var closedCells = [];
+	var rowClosedCells = [];
+	for (var i = 0; i < field.length; i++){
+		rowClosedCells = field[i].filter(function(obj) {
+
+			return !obj.opened;
+		});
+		closedCells = closedCells.concat(rowClosedCells);
+
+	}
+	if (closedCells.length === minesCount ) {
+		var result = {
+			message: "you won! congratulations!",
+			messageClass: "won",
+			mineCellClass: "flag"
+		}
+		endOfGame(field, result);
+	}
+}
```

Comment:
I like your code! Some quite complex constructions has been implemented by you. However sometimes it looks more complex than it should be. 
Coding style is good and consistent.
The game looks really nice and handy. Both style and usability are great!

There are no  tests attached to the project.